### PR TITLE
chore: improve metrics/logs for checkUserAccess/checkAdminAccess

### PR DESCRIPTION
**Pull request by enzomerca**
**What problem is this solving?**

Improve logging and metrics on checkUserAccess and checkAdminAccess directives. This will help us understand how these APIs are used.

This change keeps previous authentication behavior, it just make sure to validate all provided tokens and emit metric before applying rules.

**How should this be manually tested?**
Tested calling APIs with each directive with multiple different authentication methods.

### CODE REVIEW

File: node/directives/checkUserAccess.ts [hunk 1]
```diff
@@ -157,7 +16,155 @@ export class CheckUserAccess extends SchemaDirectiveVisitor {
       context: Context,
       info: any
     ) => {
-      await checkUserOrAdminTokenAccess(context, field.astNode?.name?.value)
```
Comment (enzomerca) [inline node/directives/checkUserAccess.ts:160]: Removed this extra function call as there was no real gain with it and to keep same style as `checkAdminAccess` directive

File: node/clients/IdentityClient.ts [hunk 2]
```diff
@@ -14,4 +14,14 @@ export default class IdentityClient extends JanusClient {
   public async validateToken({ token }: { token: string }): Promise<any> {
     return this.http.post('/api/vtexid/credential/validate', { token })
   }
+
+  public async getToken({
```
Comment (enzomerca) [inline node/clients/IdentityClient.ts:18]: The storefront-permissions app was not properly handling appkey/apptokens. Now we can get a token from identity for a given appkey/apptoken and allow/block requests properly.

File: node/directives/checkAdminAccess.ts [hunk 3]
```diff
@@ -16,68 +16,116 @@ export class CheckAdminAccess extends SchemaDirectiveVisitor {
...
-        vtex: { adminUserAuthToken, logger },
+        vtex: { adminUserAuthToken, storeUserAuthToken, logger },
         clients: { identity },
       } = context
 
+      // check if has admin token and if it is valid
+      const hasAdminToken = !!adminUserAuthToken
+      let hasValidAdminToken
+      // this is used to check if the token is valid by current standards
+      let hasCurrentValidAdminToken = false
```
Comment (enzomerca) [inline node/directives/checkAdminAccess.ts:27]: this `hasCurrentValidAdminToken` is the variable we use to track/apply original behavior. In the future we should stop using it and remove this variable.

File: node/directives/checkAdminAccess.ts [hunk 4]
```diff
@@ -16,68 +16,116 @@ export class CheckAdminAccess extends SchemaDirectiveVisitor {
...
+          hasStoreToken,
         },
-        'CheckAdminAccess'
+        'CheckAdminAccessAudit'
       )
 
-      if (!adminUserAuthToken) {
-        metric.error = 'No admin token provided'
-        sendAuthMetric(logger, metric)
+      sendAuthMetric(logger, auditMetric)
```
Comment (enzomerca) [inline node/directives/checkAdminAccess.ts:99]: we validate all possible tokens above so we can emit a metric here with all the necessary data, and below we actually apply the validations and allow/block the request.

**Review by Rudge — APPROVED** [review]

**Pull request merged**